### PR TITLE
Add item form and list that updates

### LINKED ISSUE #4: 4. As a user, I want to add a new item to my shopping list so I can start recording purchases.
A shopping list item consists of the following data points:
* Name of item
* How soon are you likely to buy it again?
   * Soon (in the next 7 days)
   * Kind of soon (in the next 14 days)
   * Not soon (in the next 30 days)
* Last purchased date
     
AC:
* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: `7` for “Soon”, `14` for “Kind of Soon”, and `30` for “Not Soon”
* The last purchased date should be set to `null` initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added

Notes:
* The add item form has the potential to be a lot of scope. The wireframe https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png has several notes related to accessibility and UX.
* Eventually, the list will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
* It's good for your team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add all of those features.
* For this story, it's sufficient just to get a simple list of the items displaying.

### PR DESCRIPTION
**Pull request by diyanamendoza**
## Description
This PR adds a form to the AddItem component that allows the user to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon). Upon form submission, the item is saved to the database, associated with the user’s token. The ItemList component was updated to render a simple, unordered list of items added.

## Related Issue

Closes #4 

## Acceptance Criteria

* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
* The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before
Screen Shot 2022-01-17 at 2 51 19 PM

### After
Screen Shot 2022-01-17 at 2 51 58 PM
Screen Shot 2022-01-17 at 2 53 51 PM

## Testing Steps / QA Criteria

Pull down the branch, add an item using the form, check the item list view for your item in the list.

### CODE REVIEW

**Review by cynthiaeddy — APPROVED** [review]
great job!

File: src/components/AddItem.js
Comment (chriskorsak) [inline src/components/AddItem.js]: I believe this is where we will want to integrate getting the token from local storage. I pulled this branch and replaced line 14 with this line on my machine:

token: localStorage.getItem('token'),

I was able to then add a new item (celery) to list and the token that was on my machine (sage window alpha) was written to the database.

File: src/components/ItemList.js
Comment (chriskorsak) [inline src/components/ItemList.js]: Same here, I believe. I also replaced line 7 on my machine with:
const token = localStorage.getItem('token');

I then navigated to the shopping list and only am seeing the item that matches my local storage token!
Screen Shot 2022-01-19 at 12 45 19 PM

**Review by chriskorsak — CHANGES_REQUESTED** [review]
Everything looks great! The comments below are essentially my take on how we might integrate the work from both pairs this week, and one possible bug. Will wait on approval though until the rest of the group weighs in on this.

**Comment by chriskorsak** [comment]
One more observation: if I first navigate to add item page and do not change 'How soon will you buy this again?' radio button, it looks like zero is written to the database for the item's 'next purchase': value. If I select another radio button, and then go back to 'soon', 7 is written to database for 'next purchase': value.

File: src/components/AddItem.js
Comment (k-monnik) [reply src/components/AddItem.js]: Thanks Chris!

**Review by chriskorsak — APPROVED** [review]
Hey team, as it stands, the changes look great! (I didn't want to hold up the review process, but in case you were to make any additional changes proposed in office hours last night, will be happy to do another review pass.)

File: src/components/ItemList.js
Comment (skylerwshaw) [inline src/components/ItemList.js:13]: This is cool! I don't recall this filtering capability on the Firebase API! Nice to not have to consume the entire list locally and filter!

File: src/components/ItemList.js
Comment (skylerwshaw) [inline src/components/ItemList.js:31]: This is 100% valid. For a fun exercise you could play with this in a javascript/browser console. Since a value of `docs.length` value of `0` will be considered _falsy_, you could omit `> 0` here since the `docs.length` of `0` will evaluate to false and have the same effect in the ternary. Again, this is 100% valid and there's a counter argument to mine that says your code is more specific/clear, taking away any ambiguity to what the ternary is comparing!

**Review by skylerwshaw — APPROVED** [review]
Awesome work incorporating the experimentation we did in Office Hours! See you all later today!

File: src/components/ItemList.js
Comment (diyanamendoza) [reply src/components/ItemList.js:31]: Always helpful to learn about the different approaches/arguments. Thx!

**Pull request merged**
